src/env.h

```diff
@@ -532,152 +532,171 @@ class AsyncRequest : public MemoryRetainer {
+  inline void ref_count_dec(uint32_t decrement);
+
+  ImmediateInfo(const ImmediateInfo&) = delete;
+  ImmediateInfo& operator=(const ImmediateInfo&) = delete;
 
-   private:
-    Environment* env_;
-  };
+ private:
+  friend class Environment;  // So we can call the constructor.
+  inline explicit ImmediateInfo(v8::Isolate* isolate);
 
-  inline size_t makecallback_depth() const;
+  enum Fields { kCount, kRefCount, kHasOutstanding, kFieldsCount };
 
-  class ImmediateInfo {
-   public:
-    inline AliasedBuffer<uint32_t, v8::Uint32Array>& fields();
-    inline uint32_t count() const;
-    inline uint32_t ref_count() const;
-    inline bool has_outstanding() const;
+  AliasedBuffer<uint32_t, v8::Uint32Array> fields_;
+};
 
-    inline void count_inc(uint32_t increment);
-    inline void count_dec(uint32_t decrement);
+class TickInfo {
+ public:
+  inline AliasedBuffer<uint8_t, v8::Uint8Array>& fields();
+  inline bool has_tick_scheduled() const;
+  inline bool has_rejection_to_warn() const;
 
-    inline void ref_count_inc(uint32_t increment);
-    inline void ref_count_dec(uint32_t decrement);
+  TickInfo(const TickInfo&) = delete;
+  TickInfo& operator=(const TickInfo&) = delete;
 
-    ImmediateInfo(const ImmediateInfo&) = delete;
-    ImmediateInfo& operator=(const ImmediateInfo&) = delete;
+ private:
+  friend class Environment;  // So we can call the constructor.
+  inline explicit TickInfo(v8::Isolate* isolate);
 
-   private:
-    friend class Environment;  // So we can call the constructor.
-    inline explicit ImmediateInfo(v8::Isolate* isolate);
-
-    enum Fields {
-      kCount,
-      kRefCount,
-      kHasOutstanding,
-      kFieldsCount
-    };
+  enum Fields { kHasTickScheduled = 0, kHasRejectionToWarn, kFieldsCount };
 
-    AliasedBuffer<uint32_t, v8::Uint32Array> fields_;
-  };
+  AliasedBuffer<uint8_t, v8::Uint8Array> fields_;
+};
 
-  class TickInfo {
-   public:
-    inline AliasedBuffer<uint8_t, v8::Uint8Array>& fields();
-    inline bool has_tick_scheduled() const;
-    inline bool has_rejection_to_warn() const;
+class TrackingTraceStateObserver :
+    public v8::TracingController::TraceStateObserver {
+ public:
+  explicit TrackingTraceStateObserver(Environment* env) : env_(env) {}
 
-    TickInfo(const TickInfo&) = delete;
-    TickInfo& operator=(const TickInfo&) = delete;
+  void OnTraceEnabled() override {
+    UpdateTraceCategoryState();
+  }
 
-   private:
-    friend class Environment;  // So we can call the constructor.
-    inline explicit TickInfo(v8::Isolate* isolate);
+  void OnTraceDisabled() override {
+    UpdateTraceCategoryState();
+  }
 
-    enum Fields {
-      kHasTickScheduled = 0,
-      kHasRejectionToWarn,
-      kFieldsCount
-    };
+ private:
+  void UpdateTraceCategoryState();
 
-    AliasedBuffer<uint8_t, v8::Uint8Array> fields_;
-  };
+  Environment* env_;
+};
+
+class ShouldNotAbortOnUncaughtScope {
+  public:
+  explicit inline ShouldNotAbortOnUncaughtScope(Environment* env);
```

Comment:
Looks like forget to format this class ?

Reply:
Thanks, fixed